### ISSUE TITLE
Fix FP S5860 (`unused-named-groups`): Consider accesses via the index syntax

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#3683]
The rule currently raises false positives when the groups of a regular expression that matches an input are accessed via the index syntax:

```typescript
const date = '02/23';

const datePattern = /(?<month>\d{2})\/(?<year>\d{2})/;
const dateMatched = date.match(datePattern);

if (dateMatched !== null) {
  checkValidity(dateMatched.groups['month'], dateMatched.groups['year']);
}
```

Reported in https://community.sonarsource.com/t/typescript-s5860-not-considering-access-via-index-syntax/82684
